drivers/sensor/vl53l0x/vl53l0x.c

```diff
@@ -378,6 +348,36 @@ static int vl53l0x_init(const struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_PM_DEVICE
+static int vl53l0x_pm_action(const struct device *dev,
+			     enum pm_device_action action)
+{
+	const struct vl53l0x_config *const config = dev->config;
+	int ret;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		ret = vl53l0x_init(dev);
```

Comment:
ok re-init the chip but shouldn't you also set xshut to 0?

Reply:
Yes, it happens in `vl53l0x_init` here: https://github.com/zephyrproject-rtos/zephyr/blob/d4522f37e87da2ba459f6056ed21ff1b0dea4faa/drivers/sensor/vl53l0x/vl53l0x.c#L327

Reply:
In short xshut will be set to 0 via the result of calling `vl53l0x_init()`.

And note we can't just set xshut to 0 at line 360 because if `CONFIG_VL53L0X_RECONFIGURE_ADDRESS` is set the device will need to go back through the address reconfiguration process.

Reply:
ok, thx for the esplanation